plugins/database/mysql/connection_producer.go

```diff
@@ -142,12 +164,40 @@ func (c *mySQLConnectionProducer) Connection(ctx context.Context) (interface{},
 		c.db.Close()
 	}
 
+	driverName := driverMySQL
+	if c.isCloud {
+		driverName = c.cloudDriverName
+
+		//@TODO - move these to init?
+		credentials := c.RawConfig["credentials"]
```

Comment:
Wondering if `credentials` is a required parameter when `auth_type` is set? If so, I suggest to move this into `Init()` and perform the validation there.

Reply:
It isn't; if you don't provide a credential, we'll attempt to use credentials stored in standard GCP credential locations, similar to https://developer.hashicorp.com/vault/docs/secrets/gcp#authentication